src/core/server/plugins/discovery/plugin_manifest_parser.ts

```diff
@@ -105,8 +103,11 @@ export async function parseManifest(
     );
   }
 
-  if (!packageInfo.dist && !isCamelCase(manifest.id)) {
-    log.warn(`Expect plugin "id" in camelCase, but found: ${manifest.id}`);
+  if (!isCamelCase(manifest.id)) {
+    throw PluginDiscoveryError.invalidManifest(
```

Comment:
This PR doesn't have an associated issue with it and it's a little difficult to determine why we're enforcing camel case now when we [didn't originally](https://github.com/elastic/kibana/pull/53759/files#diff-3a59d77fcb2057189a459799ffe416a5885a09ff4d82848c9457289b26540a7eR113-R116). If we're wrapping up the platform migration work (we had to be more lenient while plugins were still migrating over), then changing from a log warning to an explicit error makes sense.

Reply:
You are right! And I couldn't find an issue stating the reasoning behind this. The only thing I could find is the documentation about the [Manifest file](https://github.com/elastic/kibana/blob/master/docs/development/core/server/kibana-plugin-core-server.pluginmanifest.md). It claims "Identifier of the plugin. Must be a string in camelCase. Part of a plugin public contract. Other plugins leverage it to access plugin API, navigate to the plugin, etc.".

I also found that the plugin-generator already creates the id as camelCased: https://github.com/elastic/kibana/blob/7b23e7cd8b20aa889a0d31181df91bf96c0f676d/packages/kbn-plugin-generator/template/kibana.json.ejs#L2